src/blockchain/shelley/certificate.js

```diff
@@ -23,19 +23,10 @@ export type PoolUpdate = {
   pool_id: PoolId,
 }
 
-export type StakeDelegation = {
-  type: 'StakeDelegation',
-  // null here signifies NonDelegated, ie un-delegate
-  pool_id: ?PoolId,
-  account: string,
-}
-
-export type OwnerStakeDelegation = {
-  type: 'OwnerStakeDelegation',
-  // null here signifies NonDelegated, ie un-delegate
-  pool_id: ?PoolId,
-  // this is not in the certificate, but was inferred from the tx
+export type StakeDelegationType = {
+  type: number,
```

Comment:
@crchemist-ip, I agree with combining the types together, but after some discussion we are agreeing with @rooooooooob that this `type` field is supposed to be a string, to be consistent with all types used in the `CertificateType` union.

So here we should change this line to:
```
type: 'StakeDelegation',
```

But then also let's add flag like `isOwnerStake` - and the parser can set that flag instead of creating separate type like before.